packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/components/buildPerPageDataObject.ts

```diff
@@ -0,0 +1,338 @@
+import type { WebVitalsResponseItem } from "studiocms-dashboard:web-vitals";
+import { checkDate } from "./checkDate";
+
+export type perPageDataEntry = {
+    pageRoute: string;
+    sampleSize: number;
+    CLS: {
+        average: number;
+        rating: string;
+    };
+    LCP: {
+        average: number;
+        rating: string;
+    };
+    INP: {
+        average: number;
+        rating: string;
+    };
+};
+
+
+function calculateClsAverage(clsValues: number[]): number {
+
+    const sum = clsValues.reduce((acc, curr) => acc + curr, 0);
+    const average = sum / clsValues.length;
+    
+    // Round average to two decimal places
+    return Math.round(average * 100) / 100;
+}
+
+const clsDataAverage = (clsData: number[]) => {
```

Comment:
I feel all these calculations should be in a global utils file instead of being redefined each time

Reply:
I will note, some of the math is slightly different for the scoring system between the two pages, but the main calculations i could move now...

Reply:
[a9192f9](https://github.com/astrolicious/studiocms/pull/125/commits/a9192f942255f31dbc708b9662a61e887239f153) done